src/shogun/kernel/GaussianKernel.cpp

```diff
@@ -31,7 +30,7 @@ CGaussianKernel::CGaussianKernel(int32_t size, float64_t w) : CDotKernel(size)
 	set_width(w);
 }
 
-CGaussianKernel::CGaussianKernel(CDotFeatures* l, CDotFeatures* r,
+CGaussianKernel::CGaussianKernel(CFeatures* l, CFeatures* r,
```

Comment:
All issues about CDotFeatures are
CGaussianKernel is a subclass of CDotKernel() and 
in the following function
```
       virtual bool CDotKernel::init(CFeatures* l, CFeatures* r)
		{
			CKernel::init(l,r);

			ASSERT(l->has_property(FP_DOT))
			ASSERT(r->has_property(FP_DOT))
			ASSERT(l->get_feature_type() == r->get_feature_type())
			ASSERT(l->get_feature_class() == r->get_feature_class())

			if ( ((CDotFeatures*) l)->get_dim_feature_space() != ((CDotFeatures*) r)->get_dim_feature_space() )
			{
				SG_ERROR("train or test features #dimension mismatch (l:%d vs. r:%d)\n",
						((CDotFeatures*) l)->get_dim_feature_space(),((CDotFeatures*) r)->get_dim_feature_space());
			}
			return true;
		}
```

Maybe explicitly asking users to pass CDotFeatures are better.
What is more, maybe we should change the `CDotFeatures()` too
`CDotKernel::CDotKernel(CFeatures* l, CFeatures* r)`